lib/nostrum/cache/guild_cache.ex

```diff
@@ -0,0 +1,82 @@
+defmodule Nostrum.Cache.GuildCache do
+  @moduledoc """
+  Functions for retrieving guild states.
+  """
+
+  alias Nostrum.Cache.Guild.GuildRegister
+  alias Nostrum.Cache.Mapping.ChannelGuild
+  alias Nostrum.Struct.Guild
+
+  import Nostrum.Struct.Snowflake, only: [is_snowflake: 1]
+
+  @type clauses :: Keyword.t() | map
+
+  @type reason ::
+          :invalid_clauses
+          | :invalid_guild_id
+          | :invalid_channel_id
+
+  @doc """
+  Retrives a single `Nostrum.Struct.Guild` from the cache via its `id`.
+
+  Returns `{:error, reason}` if no result was found.
+
+  ## Examples
+
+  ```Elixir
+  iex> Nostrum.Cache.GuildCache.get(0)
+  {:ok, %Nostrum.Struct.Guild{id: 0}}
+
+  iex> Nostrum.Cache.GuildCache.get(10)
+  {:error, :invalid_guild_id}
+  ```
+  """
+  @spec get(Guild.id()) :: {:ok, Guild.t()} | {:error, reason}
+  def get(id) when is_snowflake(id) do
```

Comment:
How do you feel about porting over the rest of the methods into this module (`transform` et. al.)? In general they're all supposed to work as "cache" methods. And then we wouldn't have documentation for getting guilds across two modules.

Reply:
I'm on the same page as you. I'll be porting over much of the old code into the new `GuildCache` module (maybe some refactors as well). This commit was more or less an early review to see if we agree on the function layout for the new `GuildCache`.

Reply:
Speaking of which, I took the opportunity to do something similar to #53 for this new module. I've changed some of the names around to better reflect how some of the functions would be used. Let me know if anything can be named better or structured better.